### PR TITLE
Tadzik/226/offline calendar

### PR DESCRIPTION
**Pull request by tadzik**

### CODE REVIEW

File: src/app/calendar-app/calendar.service.ts [hunk 1]
```diff
@@ -38,13 +42,23 @@ export class CalendarService {
...
+            this.events = JSON.parse(cache)['events'].map(e => new RunboxCalendarEvent(e));
+            this.eventSubject.next(this.events);
+        }
+        this.calendarSubject.subscribe(_ => { this.saveCache(); });
+        this.eventSubject.subscribe(_ => {
+            this.saveCache();
+            this.lastUpdate = moment();
+        });
+
+        this.syncInterval = setInterval(() => {
```
[the remark this answers is not in the document]
Comment (tadzik) [reply src/app/calendar-app/calendar.service.ts]: Thanks; fixed in fb44dec.

File: src/app/calendar-app/calendar-app.component.ts [hunk 2]
```diff
@@ -124,6 +126,11 @@ export class CalendarAppComponent {
             this.updateEventColors();
             this.filterEvents();
         });
+
+        // force re-render to update the last update string
+        this.viewRefreshInterval = setInterval(() => {
```
[the remark this answers is not in the document]
Comment (tadzik) [reply src/app/calendar-app/calendar-app.component.ts]: Thanks; fixed in fb44dec.

File: src/app/calendar-app/calendar.service.ts [hunk 3]
```diff
@@ -38,13 +42,23 @@ export class CalendarService {
...
         private rmmapi:   RunboxWebmailAPI,
     ) {
-        console.log('Fetching calendars');
-        this.rmmapi.getCalendars().subscribe(calendars => {
-            this.calendars = calendars;
-            console.log('Calendars loaded:', calendars);
-            this.calendarSubject.next(calendars);
-            this.reloadEvents();
-        }, e => this.apiErrorHandler(e));
+        const cache = localStorage.getItem('caldavCache');
```
[the remark this answers is not in the document]
Comment (tadzik) [reply src/app/calendar-app/calendar.service.ts]: Migrated it to StorageService which uses per-uid storage.

**Comment by tadzik** [comment]
Need @petersalomonsen's acceptance on this :)

**Comment by gtandersen** [comment]
Thanks @petersalomonsen!

**Pull request merged**